File: Swift/6 kyu/6kyu_Linked_Lists_-_Remove_Duplicates.swift
class Node {
    var data: Int
    var next: Node?
    init(_ data: Int) {
        self.data = data
    }
}

func removeDuplicates(head: Node?) -> Node? {
    var current = head

    while current != nil && current!.next != nil {
        if current!.data == current!.next!.data {
            current!.next = current!.next!.next
        } else {
            current = current!.next
        }
    }
    return head
}

func printLinkedList(_ head: Node?) {
    var current = head
    while current != nil {
        print(current!.data, terminator: " -> ")
        current = current?.next
    }
    print("null")
}

File: Python/6 kyu/6kyu_Linked_Lists_-_Remove_Duplicates.py
class Node(object):
    def __init__(self, data):
        self.data = data
        self.next = None

def remove_duplicates(head):
    if not head:
        return None

    current = head
    while current.next:
        if current.data == current.next.data:
            current.next = current.next.next
        else:
            current = current.next

    return head